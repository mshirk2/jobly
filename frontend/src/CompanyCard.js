import React from "react";
import { Link } from "react-router-dom";

function CompanyCard({handle, name, description, logoUrl}) {
    return (
        <Link className="CompanyCard" to={`/companies/${handle}`}>
            <div>
                <img src={logoUrl} alt={name}/>
                <h5>{name}</h5>
                <p><small>{description}</small></p>
            </div>
        </Link>
    );
}

export default CompanyCard;