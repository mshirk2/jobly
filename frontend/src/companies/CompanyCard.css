.CompanyCard {
    max-height: 250px;
    margin-bottom: 1rem;
    color: inherit;
    border-radius: 10px;
    border: none;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);

}

.card-body {
    text-align: left;
}

.logo {
    width: 100px;
    float: left;
    margin-right: 1em;
  }

.link {
    color: black;
}