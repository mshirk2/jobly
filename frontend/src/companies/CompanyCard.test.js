import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it('renders without crashing', () =>{
    render(
        <MemoryRouter>
            <CompanyCard />
        </MemoryRouter>
    )
});

it ("matches snapshot", () =>{
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard 
                handle="test-handle"
                name="Test Company"
                description="This is a test"
            />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
});