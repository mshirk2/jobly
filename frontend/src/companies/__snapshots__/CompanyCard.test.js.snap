// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard card"
  >
    <div
      class="card-body"
    >
      <a
        class="link"
        href="/companies/test-handle"
      >
        <img
          alt="Test Company"
          class="logo"
          src="placeholderLogo.png"
        />
        <h5
          class="card-title mt-2"
        >
          Test Company
        </h5>
      </a>
      <p>
        <small>
          This is a test
        </small>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
